test/api.mocha.coffee

```diff
@@ -257,15 +265,129 @@ describe "API", ->
+              type: "guild"
+              privacy: "private"
+            ).end (res) ->
+              expectCode res, 200
+              guild = res.body
+              expect(guild.members.length).to.be 1
+              expect(guild.leader).to.be user._id
+              #Add members to guild
+              async.waterfall [
+                (cb) ->
+                  registerManyUsers 15, cb
+
+                (_members, cb) ->
+                  members = _members
+
+                  joinGuild = (member, callback) ->
+                    request.post(baseURL + "/groups/" + guild._id + "/join")
+                      .set("X-API-User", member._id)
+                      .set("X-API-Key", member.apiToken)
+                      .end ->
+                        callback(null, null)
+
+                  async.map members, joinGuild, (err, results) -> cb()
+
+              ], done
+
+          it "includes user in private group member list when user is a member", (done) ->
+            #Verfiy that when a user query's for a group they are in the group if they are a member
+            request.get(baseURL + "/groups/" + guild._id)
+            .send()
+            .end (res) ->
+              g = res.body
+              userInGroup = _.find g.members, (member) -> return member._id == user._id
+              expect(userInGroup).to.not.be undefined
+              done()
+
+          it "excludes user from viewing private group member list when user is not a member", (done) ->
+            #Remove user from group
+            request.post(baseURL + "/groups/" + guild._id + "/leave").send(
+            ).end (res) ->
+              request.get(baseURL + "/groups/" + guild._id)
+              .send()
+              .end (res) ->
+                expect res, 404
+                done()
+
+        describe "Public Guilds", ->
+          guild = undefined
+          before (done) ->
+            request.post(baseURL + "/groups").send(
+              name: "TestPublicGroup"
+              type: "guild"
+              privacy: "public"
+            ).end (res) ->
+              expectCode res, 200
+              guild = res.body
+              expect(guild.members.length).to.be 1
+              expect(guild.leader).to.be user._id
+              #Add members to guild
+              async.waterfall [
+                (cb) ->
+                  registerManyUsers 15, cb
+
+                (_members, cb) ->
+                  members = _members
+
+                  joinGuild = (member, callback) ->
+                    request.post(baseURL + "/groups/" + guild._id + "/join")
+                      .set("X-API-User", member._id)
+                      .set("X-API-Key", member.apiToken)
+                      .end ->
+                        callback(null, null)
+
+                  async.map members, joinGuild, (err, results) -> cb()
+              ], done
+
+          it "includes user in public group member list when user is a member", (done) ->
```

Comment:
The test is unreliable, sometimes it passes, sometimes it doesn't. I think because of a race condition for when users are added to the guild. Set it to skip for now.